src/core/validation/validation.pipe.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ValidationPipe as BaseValidationPipe,
+  Injectable,
+} from '@nestjs/common';
+import { ModuleRef } from '@nestjs/core';
+import { useContainer, ValidatorOptions } from 'class-validator';
+import { ValidationException } from './validation.exception';
+
+@Injectable()
+export class ValidationPipe extends BaseValidationPipe {
+  constructor(private readonly moduleRef: ModuleRef) {
+    super({
+      transform: true,
+      skipMissingProperties: true,
+      exceptionFactory: (es) => new ValidationException(es),
+    });
+  }
+  private readonly containerForLib = {
+    get: (type: any) => {
+      if (type.name === 'CustomConstraint') {
+        // Prototype-less constraints. Null to fall back to default, which just calls constructor once.
+        return null;
+      }
```

Comment:
These types are created dynamically created classes, so we don't have a way to reference/provide them in a nestjs module.
We return null here, and have `fallback: true`, so that the library does the default thing of memoized `new type()`.
These are from built in constraints, or ours like this https://github.com/SeedCompany/cord-api-v3/blob/ccd94582658b53c1e8904e7d27e33786683d7dfa/src/common/validators/email.validator.ts#L13-L16

So they don't need to be injected. Actual custom constraint classes, will have a static class ref, that will be provided to a nestjs module, and used here.

I also have `fallbackOnErrors: false` (as default), so we should get errors when asking for a constraint that hasn't be provided/registered.